cloudferrylib/os/storage/cinder_storage.py

```diff
@@ -60,6 +62,20 @@ def get_client(self, params=None):
             insecure=self.config.cloud.insecure
         )
 
+    def get_endpoint(self):
+        return self.identity_client.get_endpoint_by_service_type(
+            service_type='volume',
+            endpoint_type='publicURL',
+        )
+
+    def get_endpoint_host(self):
+        return self.extract_host(self.get_endpoint())
+
+    @staticmethod
+    def extract_host(url):
+        m = re.search('//([^:^/]*)', url)
```

Comment:
I think, we can use a compiled object, like:

    RE_EXRACT_HOST = re.compile('//([^:^/]*)')
...

    m = RE_EXRACT_HOST.search(url)